### PR TITLE
fix(material/button): fix outlined button focus color

### PR DESCRIPTION
**Pull request by wagnermaciel**

### CODE REVIEW

File: src/material/button/_button-theme-private.scss [hunk 1]
```diff
@@ -19,6 +19,7 @@
     opacity: map.get($opacities, hover);
   }
 
+  &:focus,
   &.cdk-program-focused,
   &.cdk-keyboard-focused {
```
Comment (crisbeto) [inline src/material/button/_button-theme-private.scss:24]: I think the point of `cdk-program-focused` and `cdk-keyboard-focused` below was not to use `:focus` which also shows up for things like mouse clicks.

Comment (wagnermaciel) [reply src/material/button/_button-theme-private.scss:24]: What triggers these classes getting appended? From what I can see, the regular/default button doesn't get these so the focus state never changes the background color

Comment (crisbeto) [reply src/material/button/_button-theme-private.scss:24]: They get added if `FocusMonitor.monitor` is called on the button and the user tabs into it or `.focus()` is called on it directly. Maybe the `FocusMonitor` call got lost somewhere in the refactors?

Comment (wagnermaciel) [reply src/material/button/_button-theme-private.scss:24]: Is the focus state not meant to show up for mouse clicks?

Comment (crisbeto) [reply src/material/button/_button-theme-private.scss:24]: Yeah that was one of the reasons we introduced the FocusMonitor.

**Pull request closed without merging**